### PR TITLE
Update FatFS to R0.15 Patch3

### PR DESCRIPTION
**Pull request by networkfusion**
We currently use R0.14 and there have been a few updates since then.

See http://elm-chan.org/fsw/ff/updates.html and http://elm-chan.org/fsw/ff/patches.html

Also incoroprates #409

### CODE REVIEW

**Comment by networkfusion** [comment]
Tested against N64Flashcart Menu with no issues encountered.

File: src/fatfs/ffconf.h [hunk 1]
```diff
@@ -66,6 +59,33 @@
 /* This option switches f_forward() function. (0:Disable or 1:Enable) */
 
 
+// FIXME: remove this, do not merge to stable without further discussion
```
Comment (rasky) [inline src/fatfs/ffconf.h]: Why are you adding this FIXME here?

Comment (networkfusion) [reply src/fatfs/ffconf.h]: I am not. That is your comment in unstable.

Comment (networkfusion) [reply src/fatfs/ffconf.h]: https://github.com/DragonMinded/libdragon/blob/352679922bad5403f8ea4b9123dfe0403d030bc2/src/fatfs/ffconf.h#L27

Comment (rasky) [reply src/fatfs/ffconf.h]: oh i had forgotten about that one. Do you still need that #ifdef?

Comment (networkfusion) [reply src/fatfs/ffconf.h]: Feel free to remove it. If needed I can workaround using SED before build.

Comment (networkfusion) [reply src/fatfs/ffconf.h]: I have removed it.

**Comment by rasky** [comment]
Also merged to trunk

**Pull request merged**